gnocchi/storage/__init__.py

```diff
@@ -688,6 +688,26 @@ def _map_compute_splits_operations(bound_timeserie):
             if metric.needs_raw_data_truncation:
                 indexer_driver.update_needs_raw_data_truncation(metric.id)
 
+            # Mark when the metric receives its latest measures
+            indexer_driver.update_last_measure_timestmap(metric.id)
```

Comment:
i'm wondering if there's a way to do this without updating indexer db if metric_inactive_after functionality is not enabled? seems like a performance hit as we're adding additional db call

Reply:
Yes, it is an additional call. However, nothing that costly. I mean, this is an async process anyways. This extra cost is worth when we see the benefits that this is bringing to the code base.

Reply:
well there's this one and then the resource lookup call below. i'm thinking if it makes sense to move the resource logic below to janitor process maybe so this step just focuses on processing measures data.

i think that way, the janitor (or something else) could technically just make a single db query which unset ended_at for all resources where ended_at not null and has metrics with last_update_timestamp > now - ttl?

if the above works, i imagine most of logic in resource_ended_at_normalization could be done as similar db query as well?

Reply:
don't know if the above was considered or it makes sense? tbh. i wrote it so long ago i don't remember if it was already concluded to be dumb/invalid.

Reply:
That was considered, but to make the code cleaner and more intuitive, we decided for this approach. If you think something else is better we could try to work together with you to change a bit the logic. However, so far, this patch is in use for over 6 months already and everything is so far so good.